[Chen Kai Yi] iP

kaiyichen opened the pull request:
# DukeTime
>“Time management is not a peripheral activity or skill. It is the core skill upon which everything else in life depends.” – Brian Tracy

DukeTime manages your time by remembering things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://github.com/kaiyichen/ip)
2. double-click it. :point_up_2:
3. add your tasks. :giraffe:
4. let it manages your tasks for you!! :satisfied:

And what's best?! It is **FREEEE!!**

Features:
- [x] Manages Tasks
- [x] Manages deadlines
- [ ] Reminder (coming soon!!) 

If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

kaiyichen committed:
Level-1

kaiyichen committed:
Level-1 corrected

kaiyichen committed:
Level-2

kaiyichen committed:
Level-3

kaiyichen committed:
Level-4(semi-done)

kaiyichen committed:
Level-4 completed

kaiyichen committed:
Level-5

kaiyichen committed:
Level-6

kaiyichen committed:
A-TextUiTesting

kaiyichen committed:
Completed Level-7

kaiyichen committed:
completed Level-8

kaiyichen committed:
Updated Level-7

kaiyichen committed:
Merge branch 'branch-Level-7'

kaiyichen committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java

kaiyichen committed:
A-MoreOOP completed (requires some checking)

kaiyichen committed:
Add classes to package

kaiyichen committed:
Add A-JUnit Tests

kaiyichen committed:
added Javadoc for all classes

kaiyichen committed:
fixed coding standard

kaiyichen committed:
added find function

kaiyichen committed:
Merge branch 'branch-A-JavaDoc'

kaiyichen committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/DeadLine.java
#	src/main/java/duke/TaskList.java

kaiyichen committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/TaskList.java

kaiyichen committed:
Merge branch 'add-gradle-support' of https://github.com/kaiyichen/ip

kaiyichen committed:
Add Gradle

weijuey commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,77 @@
+     * @param list_of_inputs
+     * @throws IOException
+     */
+    public void reSavingFiles (ArrayList<Task> list_of_inputs) throws IOException {
```
list_of_inputs should be in camelCase. Would it also be better to name the argument as simply `inputs`?

ngjunkang replied:
Just to add on to @weijuey, resaving seems to be a correct word, so maybe you can consider `resavingFiles`?

weijuey commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,142 @@
+    public void mark( int curr_no) throws IOException {
+        Task task_done = listOfInputs.get(curr_no);
+        System.out.println("Good work!! I have marked it done:\n" + task_done.markedDone());
+        storage.updateData(task_done.message(),filePath);
```
There should be a space after commas. Noted it in other places as well.

weijuey commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,142 @@
+        }
+    }
+
+    public void mark( int curr_no) throws IOException {
```
There should be no space after `(`. Noted it in other places as well.

ngjunkang replied:
`curr_no` should be in camelCase as well according to coding standard, such as `currNo` or `taskIndex`. Noted the same issue in other places as well.

weijuey commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,142 @@
+    /**
+     * to update on list and display the deadline message once it is called
+     * @param str
+     * @return
```
Did you forget to fill in the return description?

kaiyichen committed:
add CheckStyles

weijuey commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,94 @@
+    }
+
+
+    String greeting = "Hello! I'm Duke\nWhat can I do for you?";
+    String separation = "\n******************************\n";
```
Did you mean to leave the access modifiers unspecified?

weijuey commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,94 @@
+ *
+ */
+
+public class Ui {
```
Nice, you didn't forget that i should not be capitalised.

weijuey commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,94 @@
+        }
+
+
+            if (inp.equals("bye")) {
```
Would a switch statement look better?

weijuey reviewed (commented):
Looks good, just needs some fixes to whitespace placement.

ngjunkang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,45 @@
+import duke.Ui;
+
+import duke.*;
+import java.io.*;
```
I remember wildcard imports (duke.* etc) are not recommended, maybe you might want to check with the coding standard again? I've noted this in other Java files as well.

In addition, perhaps you might want to put all your Java files in a package called duke?

ngjunkang commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,77 @@
+        }
+    }
+
+    /**
+     * read data from file
+     * @param filePath
+     */
```
For Javadoc method header comments, maybe you can consider the following format as stated in the coding standard:
```java
/**
 * Reads data from file.
 *
 * @param filePath the file path to read the data from.
 */
```
I noticed similar issues (missing description, no starting verb, starting verb is non-compliant, no link break between description and `@param`, no description for `@param`, `@return`) in several other places too.

ngjunkang commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,142 @@
+        String[] event = str.split("/at ", 2);
+        Event eventTemp = new Event(event[0], event[1]);
+        listOfInputs.add(eventTemp);
+        storage.updateData(eventTemp.message(),filePath);
```
As stated in the coding standards, commas should be followed by a space. I've noted similar issues (comma preceded by a space etc) in other places as well, but it should be as follows: `f(arg1, arg2)` for any method/constructor `f` with arguments `arg1` and `arg2`, what do you think?

ngjunkang commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,62 @@
+     */
+    public String markedDone() {
+        this.isDone = true;
+        return  this.message();
```
Did you forget to remove the additional space? Noted it in other places as well.

ngjunkang reviewed (commented):
LGTM, just some nits to fix, good job! 👍

kaiyichen committed:
Merge branch 'branch-A-CheckStyle'

lawwm commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,52 @@
+        }
+    }
+
+    public static void main(String[] args) throws DukeException, IOException {
+        new Duke("C:\\DukeDirectory", "C:\\DukeDirectory\\DukeSave.txt").run();
+    }
```
Maybe you can avoid magic constants, and name the string?

lawwm commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,155 @@
+    private String directory;
+    private String filePath;
+
+    private Storage storage = new Storage(directory, filePath);
+    /**
+     * @param directory
+     * @param filePath
+     */
+    public TaskList (String directory, String filePath) {
+        this.directory = directory;
+        this.filePath = filePath;
+
+    }
```
I think you should initialize storage field within constructor since it uses the parameters of constructor?

lawwm commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,155 @@
+        int currNo = Integer.parseInt(str) - 1;
+        listOfInputs.remove(currNo);
+        Delete deleteTemp = new Delete(listOfInputs.get(currNo).description);
+        System.out.println(deleteTemp.message() + "\nYou have " + listOfInputs.size() + " tasks in the list.");
+    }
```
I think that direct output to UI should be encapsulated within the UI class?

lawwm commented on src/main/java/duke/Todos.java:
```diff
@@ -0,0 +1,28 @@
+/**
+ * todo class
+ * inherit from Task class
+ */
+public class Todos extends Task {
+
```
Since Todos is a single instance of Task, maybe it should be named Todo?

lawwm commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,74 @@
+     * @param descrip
+     * @throws DukeException
+     */
+    public void invalidDeadline (String descrip) throws DukeException {
```
Maybe change method name to verb?

lawwm reviewed (commented):
Seems good to me. Code is easy to read, so I only suggest some minor changes.

chriswengqi commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,97 @@
+        }
+
+
+        if (inp.equals("bye")) {
+            isExit = true;
+            storage.reSavingFiles(taskList.listOfInputs);
+            System.out.println("Bye. Hope to see you again soon!");
+        } else if (inp.equals("list")) {
+            taskList.list();
+
+        } else if (temp[0].equals("mark")) {
+            int currNo = Integer.parseInt(temp[1]) - 1;
+            taskList.mark(currNo);
+        } else if (temp[0].equals("unmark")) {
+            int currNo = Integer.parseInt(temp[1]) - 1;
+            taskList.unMark(currNo);
+        } else if (temp[0].equals("todo")) {
+            System.out.println(taskList.toDo(temp[1]));
+        } else if (temp[0].equals("deadline")) {
+            System.out.println(taskList.deadLine(temp[1]));
+        } else if (temp[0].equals("event")) {
+            System.out.println(taskList.event(temp[1]));
+        } else if (temp[0].equals("delete")) {
+            // DeleteCommand d = new DeleteCommand(DUKE_DIRECTORY,DUKE_TXTFILE);
+            taskList.delete(temp[1]);
+        } else if (temp[0].equals("find")) {
+            taskList.find(temp[1]);
+        }
+
+
+    }
+
```
Can consider using switch statements instead of if-else.

chriswengqi commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,155 @@
+    /**
+     * to update on the list and display the event message once it is called
+     * @param str
+     * @return
```
I think you should fill in the return statement

chriswengqi commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,155 @@
+ */
+public class TaskList {
+
+    ArrayList<Task> listOfInputs = new ArrayList<>();
```
Did you forget to declare class variable scope?

chriswengqi reviewed (commented):
LGTM, just a few minor changes needed to improve code readability!

kaiyichen committed:
complete level-10

kaiyichen committed:
Merge branch 'branch-Level-10'

kaiyichen committed:
Deadline, DukeException and Event class: add assert

Deadline, DukeException and Event have some assumptions

Add assert function into Deadline, DukeException and  Event to send exception once the assumption is not made. For example, input is "deadline task by 1200" but the code is assuming there will be command "/by".

Using assertion can test my assumptions and will throw an error is assumption is not made true.

kaiyichen committed:
Add Code Quality to all classes

utilize code quality to improve code's understandability

avoid complicated expressions by defining them as different variables. Deleted dead code to avoid confusions.

To improve understandability of the code.

kaiyichen committed:
Merge pull request #1 from kaiyichen/branch-A-Assertions

Deadline, DukeException and Event class: add assert

kaiyichen committed:
Merge branch 'master' of https://github.com/kaiyichen/ip into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Ui.java

kaiyichen committed:
Merge branch 'branch-A-Assertions'

kaiyichen committed:
Merge branch 'master' of https://github.com/kaiyichen/ip

kaiyichen committed:
add C-BetterSearch

kaiyichen committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java

kaiyichen committed:
Add User guide, Ui.png and fixed some errors

kaiyichen committed:
Add JAR file

kaiyichen committed:
fixed file not found error and added exception to check on date. Fixed search method too

kaiyichen committed:
Added more exception handling for find and delete

kaiyichen committed:
fixed to pass smoke test
